python-checks/src/test/resources/checks/argumentNumber.py

```diff
@@ -214,3 +214,20 @@ def logging_api():
 def foo(day, tz):
     b = datetime.date.fromordinal(day).replace(tzinfo=tz) # FN SONARPY-1472
     a = datetime.datetime.fromordinal(day).replace(tzinfo=tz) # OK
+
+
+def bound_and_unbound_methods():
+    class ClassWithMethod:
+        def some_method(self, param): ...
+
+    unbound_method = ClassWithMethod.some_method
+    bound_method = ClassWithMethod().some_method
+
+    unbound_method(1, 2) # OK
+    bound_method(1) # OK
+
+    unbound_method() # FN
+    unbound_method(1) # FN
+    unbound_method(1, 2, 3) # FN
+    bound_method() # FN
+    bound_method(1, 2) # FN
```

Comment:
IMO It is still worth it to add the reference. We can still update the issue number when we tackle SONARPY-2285 to point to the next ticket. I assume that SONARPY-2285 and the other required ticket will be linked so it will be clear enough.
I'm not super strong on it so feel free to keep it like this ;)

Reply:
Fair enough, I made the change then!